cogs/respects/core.py

```diff
@@ -157,29 +168,31 @@ async def payRespects(self, ctx: Context):
                 finally:
                     chData[KEY_MSG] = None
 
-            confUserIds = chData[KEY_USERS]
-            currentGuild: discord.Guild = ctx.guild
-            members = list(
+            confUserIds: List[int] = chData[KEY_USERS]
+            currentGuild: Optional[Guild] = ctx.guild
+            members: List[Optional[Member]] = list(
```

Comment:
1. I think this can still be `currentGuild: Guild = ...` since all commands that invoke this have `guild_only` decorators so `ctx.guild` must exist.
2. Can this just be `List[Member]`, given that `filter` will exclude all `NoneType` objects that are returned by `get_member(...)`?

Reply:
Yeah, will fix.

Reply:
1. The annoying thing about `Context.guild` is that it is `None`-able by type checking but cannot be `None` in runtime. @Injabie3 what do you think about adding `assert`s in this case? We can basically say that `ctx.guild` cannot be `None`, else the cog crashes.

2. `filter` cannot tell that the function will eliminate all "false" elements (basically `None`s in this case). We could do a `typing.cast()` to assume the return type, but I think a more intuitive approach is just to... not use `filter` and either do an inline `for` loop or regular loops that add elements to `members`.

Reply:
1. `assert` wfm.
2. TIL, either or wfm.

Reply:
About these two and the `oldRespectTime` one tatsu caught below, if we're going to use a type checker then optimizing for it makes sense. But otherwise aligning with runtime types seems intuitive. Also related to what we talked about on discord, right now it's best effort type hints. We could change that and force everyone to use a type checker (just like with the `black` formatter) and that will ensure consistency across all cogs. Need to decide.

Reply:
Yeah it's fine. We can always reiterate the code after this PR for more type checking-related refactoring if we want. Perhaps we can incorporate something like [this](https://github.com/microsoft/pyright/blob/main/docs/ci-integration.md) when we've got our cogs restructured and with some adequate type hinting in the code.

Reply:
Ok, then for the time being I'm just going to make sure these are aligned to runtime types. And revisit this in a separate PR after we adopt Pyright.